### PR TITLE
Fix manage page admins list

### PR DESCRIPTION
**Pull request by rbennettcw**
## Link to Issue
Closes: #4045
Closes: #3985

## Description of Changes
- Adds a new `/roles` endpoint to correctly list roles by chain ID and permissions filter
- Remove deprecated getRoles endpoint
- Updated UI to use updated (non-external) URL

## Test Plan
- Navigate to the Manage page of a community you are an admin/mod for– confirm that the admins list renders as expected
- Click the `X` next to a mod/admin and attempt to remove them– confirm that the user is converted to a member

## Deployment Plan
N/A

## Other Considerations
- With there being some deprecated tables/endpoints, it would be great to have a guide on how to correctly set myself as admin to a community in the local environment. I *thought* I did it properly before, but apparently not.
- I can confirm that a request to `http://localhost:8080/api/roles?chain_id=timelessfinance&permissions[]=moderator&permissions[]=admin` returns the admins for that chain, so it *should* render in the UI properly.

### CODE REVIEW

File: packages/commonwealth/server/routing/external.ts [hunk 1]
```diff
@@ -151,7 +151,6 @@ export function addExternalRoutes(
     deleteEntities.bind(this, 'chain_id', models, models.Topic)
   );
 
-  router.get('/roles', getRolesValidation, getRoles.bind(this, models));
```
Comment (jnaviask) [inline packages/commonwealth/server/routing/external.ts:154]: Feel free to delete this route entirely. We can also remove it from swagger_external.json, so it doesn't display on our route docs.

**Comment by jnaviask** [comment]
As for "how to correctly set myself as admin to a community in the local environment."

1. Set yourself as god mode / site admin by changing your user flag in the database ("isAdmin").
2. Go into the manage community page on the community you want to set yourself as admin for.
3. Search yourself in the upgrade roles form and select it.
4. Select the "admin" radio button and hit "upgrade member"
5. Remove yourself as god mode / site admin in the database.

At this point, you will be a community admin.

**Comment by rbennettcw** [comment]
> @rbennettcw As for "how to correctly set myself as admin to a community in the local environment."
> 
> 1. Set yourself as god mode / site admin by changing your user flag in the database ("isAdmin").
> 2. Go into the manage community page on the community you want to set yourself as admin for.
> 3. Search yourself in the upgrade roles form and select it.
> 4. Select the "admin" radio button and hit "upgrade member"
> 5. Remove yourself as god mode / site admin in the database.
> 
> At this point, you will be a community admin.

Awesome! I didn't know about god mode.

**Comment by rbennettcw** [comment]
Applied feedback.

**Review by jnaviask — APPROVED** [review]
Route works, but would be great if we could quickly fix up the state issues with adding/removing admins/mods. Both cases require refreshes to update the page.

**Review by CowMuon — APPROVED** [review]
LGTM

**Comment by CowMuon** [comment]
Already merged but just to note we could use some test coverage here. @rbennettcw @jnaviask

**Pull request merged**